### PR TITLE
MAIN - adds classes for icon on left

### PR DESCRIPTION
**Pull request by paschalidi**

### CODE REVIEW

File: src/Input/input.style.js [hunk 1]
```diff
@@ -22,9 +23,16 @@ const InputStyles = styled.div`
...
     font-weight: normal;
     color: #353f41;
-    padding: 2px 16px;
     transition: border-color 0.2s ease, box-shadow 0.2s ease;
 
+    &.default-padding {
+      padding: 2px 16px;
+    }
+
+    &.extra-padding {
```
Comment (herman-rogers) [inline src/Input/input.style.js]: Instead of extra why not icon-padding? That way we know what it is for.

File: src/Input/input.style.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const InputStyles = styled.div`
   width: 100%;
 
   label {
+    position: relative;
```
Comment (herman-rogers) [inline src/Input/input.style.js]: Make sure this is responsive and looks good on small and large screens

File: src/Input/input.style.js [hunk 3]
```diff
@@ -22,9 +23,16 @@ const InputStyles = styled.div`
     font-size: 18px;
     font-weight: normal;
     color: #353f41;
-    padding: 2px 16px;
     transition: border-color 0.2s ease, box-shadow 0.2s ease;
 
+    &.default-padding {
```
Comment (herman-rogers) [inline src/Input/input.style.js]: Also, why not just put this in the parent class and override the style with the icon padding inside that class

File: src/Input/input.style.js (hunk 1 above)
Comment (herman-rogers) [reply src/Input/input.style.js]: Also is the & required?

Comment (paschalidi) [reply src/Input/input.style.js]: & is required yes.

File: src/Input/input.style.js (hunk 3 above)
Comment (paschalidi) [reply src/Input/input.style.js]: i dont get that. please elaborate. also do you see something really wrong with the current implementation ?

Comment (paschalidi) [reply src/Input/input.style.js]: sorry miss-clicked resolve

Comment (herman-rogers) [reply src/Input/input.style.js]: You don’t need the default class it’s redundant. This will also simplify the login in the component.

Comment (paschalidi) [reply src/Input/input.style.js]: I thought it helps with readability. But but yeah I will go ahead

File: src/Input/input.style.js (hunk 2 above)
Comment (paschalidi) [reply src/Input/input.style.js]: done

File: src/Input/input.style.js (hunk 1 above)
Comment (paschalidi) [reply src/Input/input.style.js]: its because it is like `input.extra-padding`

File: src/Input/Input.jsx [hunk 4]
```diff
@@ -25,13 +38,15 @@ Input.propTypes = {
...
 };
 
 Input.defaultProps = {
   type: "",
   name: "default",
   placeholder: "",
   label: "",
-  optional: false
+  optional: false,
+  iconName: undefined
```
Comment (herman-rogers) [inline src/Input/Input.jsx]: Make this “” instead of undefined. Empty strings are falsy. It’s just a bit more understandable.

**Review by herman-rogers — APPROVED** [review]

**Pull request merged**